src/test/ui/consts/control-flow/interior-mutability.rs

```diff
@@ -21,7 +22,22 @@ const Y: Option<Cell<i32>> = {
     y
 };
 
+const Z: Option<Cell<i32>> = {
+    let mut z = None;
+    let mut i = 0;
+    while i < 10 {
+        if i == 8 {
+            z = Some(Cell::new(4));
+        }
+
```

Comment:
Assign `None` for i == 9 to show that we only look at static information and not the way `i ` behaves during execution